### DISCUSSION TITLE
Multiple matching filter

### DISCUSSION THREAD

**Discussion by ivanradojevic-web-dev** [spatie/laravel-query-builder#549] [Q&A]
How to make multiple matching filter from docs /users?filter[name]=seb,freek
```
@foreach(\App\User::get()->pluck("name")->unique() as  $name)
<a href="{{ route('products.index', array_merge(request()->query(),  [ 'filter[name]' => $name  ] ) )  }}" >{{$name}}</a>
@endforeach
```

**Comment by AlexVanderbist** [comment]
Hi, I'm not sure what you mean. What is the request url you want to use and what do you want it to do?

**Reply by ivanradojevic-web-dev** [reply]
Maybe this as example from docs /users?filter[name]=seb,freek . How to make links (not from url) in blade to be able to accept the array as in the example seb,freek?

**Reply by ivanradojevic-web-dev** [reply]
I had another question, which may be a bug. Nobody answered me.

Spatie example: has a bug to show same (double) key, not ok...  ?filter%5Bname%5D=adidas&filter%5Bname%5D=puma
$products = QueryBuilder::for(Product::class)
    		->allowedFilters('name', AllowedFilter::exact('size') )
    		->get();

Laravel raw example show only last "clicked" key, that is okay... ?filter%5Bname%5D=adidas 
$name = $request->input('name');
$size = $request->input('size');
        $products= Product::when($request->get('name'), function ($query, $name) {
        	return $query->where('name', $name);
    	})->when($request->get('size'), function ($query, $size) {
        	return $query->where('size', $size);
   	    })->get();

**Reply by AlexVanderbist** [reply]
I think you were almost there for your initial question. Something like this will make a link that has all the users in the filter:

```php
@php
$users = \App\User::get()->pluck("name")->unique();
@endphp

<a href="{{ route('products.index', array_merge(request()->query(),  [ 'filter[name]' => compact($users)  ] ) )  }}" >Filter all users</a>
```

What you describe as a bug is actually intended by the HTTP spec: having the same query parameter twice will overwrite the first one.

Good luck!

**Reply by ivanradojevic-web-dev** [reply]
With this code, I dont have any query string.
